lib/devmapper_mock.ml

```diff
@@ -324,68 +197,101 @@ module DeviceSet = struct
-  try
-    ref (DeviceSet.load_file !persistent_fn)
-  with _ ->
-    ref (DeviceSet.make ())
+  let create device targets =
+    DeviceSet.create !devices device targets;
+    DeviceSet.save_file !devices !persistent_fn
 
-let create device targets =
-  DeviceSet.create !devices device targets;
-  DeviceSet.save_file !devices !persistent_fn
+  let remove device =
+    DeviceSet.remove !devices device;
+    DeviceSet.save_file !devices !persistent_fn
 
-let remove device =
-  DeviceSet.remove !devices device;
-  DeviceSet.save_file !devices !persistent_fn
+  let reload device targets =
+    DeviceSet.reload !devices device targets;
+    DeviceSet.save_file !devices !persistent_fn
 
-let reload device targets =
-  DeviceSet.reload !devices device targets;
-  DeviceSet.save_file !devices !persistent_fn
+  let suspend device =
+    DeviceSet.suspend !devices device;
+    DeviceSet.save_file !devices !persistent_fn
 
-let suspend device =
-  DeviceSet.suspend !devices device;
-  DeviceSet.save_file !devices !persistent_fn
+  let resume device =
+    DeviceSet.resume !devices device;
+    DeviceSet.save_file !devices !persistent_fn
 
-let resume device =
-  DeviceSet.resume !devices device;
-  DeviceSet.save_file !devices !persistent_fn
+  let mknod device path mode =
+    DeviceSet.mknod !devices device path mode;
+    DeviceSet.save_file !devices !persistent_fn
 
-let mknod device path mode =
-  DeviceSet.mknod !devices device path mode;
-  DeviceSet.save_file !devices !persistent_fn
+  let stat device =
+    let ret = DeviceSet.stat !devices device in
+    DeviceSet.save_file !devices !persistent_fn;
+    ret
 
-let stat device =
-  let ret = DeviceSet.stat !devices device in
-  DeviceSet.save_file !devices !persistent_fn;
-  ret
+  let ls () =
+    let ret = DeviceSet.ls !devices () in
+    DeviceSet.save_file !devices !persistent_fn;
+    ret
 
-let ls () =
-  let ret = DeviceSet.ls !devices () in
-  DeviceSet.save_file !devices !persistent_fn;
-  ret
+  let clear () =
+    DeviceSet.clear !devices ();
+    DeviceSet.save_file !devices !persistent_fn
 
-let clear () =
-  DeviceSet.clear !devices ();
-  DeviceSet.save_file !devices !persistent_fn
+  let get_persistent () =
+    !persistent_fn
 
-let get_persistent () =
-  !persistent_fn
+  let set_persistent filename =
+    persistent_fn := filename
 
-let set_persistent filename =
-  persistent_fn := filename
+  let save_file path =
+    DeviceSet.save_file !devices path
 
-let save_file path =
-  DeviceSet.save_file !devices path
+  let load_file path =
+    devices := DeviceSet.load_file path
+end
+
+let create = Mock.create
+let remove = Mock.remove
+let reload = Mock.reload
+let suspend = Mock.suspend
+let resume = Mock.resume
+let mknod = Mock.mknod
+let stat = Mock.stat
+let ls = Mock.ls
+let clear = Mock.clear
+let get_persistent = Mock.get_persistent
+let set_persistent = Mock.set_persistent
+let save_file = Mock.save_file
+let load_file = Mock.load_file
+
+module type DM_intf = sig
```

Comment:
@djs55, thanks for comments. To enable the clients to choose which implementations in run time, should the module type `S.DEVMAPPER` be exposed too? Where should the common module type be placed in common OCaml convention?

Also, as submodules (e.g., `Location`, `Striped` and `Target`) exist within `Devmapper.Linux`, I think they are common to `Devmapper.Linux` and `Devmapper.Mock`. Where should they go then?